extensions/pkg/controller/worker/genericactuator/actuator_reconcile.go

```diff
@@ -309,55 +324,115 @@ func (a *genericActuator) deployMachineDeployments(ctx context.Context, cluster
 
 // waitUntilWantedMachineDeploymentsAvailable waits until all the desired <machineDeployments> were marked as healthy /
 // available by the machine-controller-manager. It polls the status every 5 seconds.
-func (a *genericActuator) waitUntilWantedMachineDeploymentsAvailable(ctx context.Context, cluster *controller.Cluster, worker *extensionsv1alpha1.Worker, wantedMachineDeployments worker.MachineDeployments) error {
-	return retryutils.UntilTimeout(ctx, 5*time.Second, 5*time.Minute, func(ctx context.Context) (bool, error) {
-		var numHealthyDeployments, numUpdated, numAvailable, numDesired, numberOfAwakeMachines int32
+func (a *genericActuator) waitUntilWantedMachineDeploymentsAvailable(ctx context.Context, cluster *controller.Cluster, worker *extensionsv1alpha1.Worker, alreadyExistingMachineDeploymentNames sets.String, alreadyExistingMachineClassNames sets.String, wantedMachineDeployments extensionsworker.MachineDeployments, clusterAutoscalerUsed bool) error {
+	rollingUpdate := false
+	autoscalerIsScaledDown := false
+	workerStatusUpdatedForRollingUpdate := false
 
-		// Get the list of all existing machine deployments
-		existingMachineDeployments := &machinev1alpha1.MachineDeploymentList{}
-		if err := a.client.List(ctx, existingMachineDeployments, client.InNamespace(worker.Namespace)); err != nil {
+	return retryutils.UntilTimeout(ctx, 5*time.Second, 3*time.Minute, func(ctx context.Context) (bool, error) {
```

Comment:
Why is it lowered to `3m`?

Reply:
The idea was to lower the amount of time that has to pass until it is checked if the MC is stuck.
The operation will be retried anyhow due to the erroneous reconciliation.

However no strong opinion here.

Reply:
If it's lowered then the `Worker`'s last operation is faster set to `Error`. I would keep it at `5m`. I think MCM being stuck is a very rare case anyways.